Restructure `issues` commands

JenySadadia opened the pull request:
Restructure KCIDB issues related commands.

Modifications:
- Remove `issues` command group and subcommand handlers as part of command restructuring
- Implement `results issue` command
    - Fetch KCIDB issue matching provided ID.  Also, get associated builds and tests.
    `kci-dev results issue --id <id> --origin <origin>`
- Implement `results issues` command
    - Fetch KCIDB issues from the dashboard
    - Get new issues for checkout with `--new` option
    - Get failed/inconclusive builds and boots without
    any associated issues with `--missing` option

JenySadadia commented:
`issue` command output:
```
$ kci-dev results issue --id redhat:issue_2753
Issue information:
- Issue ID: redhat:issue_2753
  origin: redhat
  version: 1716387967
  comment: KUNIT: KFENCE failed to allocate
  field_timestamp: 2025-08-06T23:44:44.050949Z
  culprit_code: True
  culprit_tool: False
  culprit_harness: False

Associated builds:
No associated builds found

Associated tests:
- redhat:1626193375-aarch64-kernel-64k_upt_18
- redhat:1633276830-aarch64-kernel-64k_upt_18
- redhat:1633379163-aarch64-kernel-64k_upt_7
- redhat:1634393276-aarch64-kernel-64k_upt_19
- redhat:1634768214-ppc64le-kernel_upt_8
- redhat:1626664281-aarch64-kernel_upt_18
- redhat:1635006205-aarch64-kernel-64k_upt_19
- redhat:1569812661-ppc64le-kernel_upt_15
- redhat:koji-126644432-aarch64-kernel-64k_upt_6
- redhat:1639467482-aarch64-kernel-64k_upt_16
- redhat:1640007736-aarch64-kernel_upt_18
- redhat:1640003284-aarch64-kernel_upt_18
- redhat:1641503170-x86_64-kernel_upt_13
- redhat:1642354951-x86_64-kernel_upt_13
- redhat:1639467482-aarch64-kernel-64k_upt_16
- redhat:1576630306-aarch64-kernel-64k_upt_18
- redhat:koji-126697389-aarch64-kernel-64k_upt_6
- redhat:1692715109-ppc64le-kernel_upt_9
- redhat:1692766765-ppc64le-kernel_upt_9
- redhat:1693711336-ppc64le-kernel_upt_9
- redhat:1696353777-ppc64le-kernel_upt_16
- redhat:1697622989-ppc64le-kernel_upt_9
- redhat:1697552504-ppc64le-kernel_upt_17
- redhat:koji-129805675-aarch64-kernel-64k_upt_6
- redhat:koji-129867125-aarch64-kernel-64k_upt_6
- redhat:koji-129867125-aarch64-kernel-64k_upt_6
- redhat:koji-129949568-aarch64-kernel-64k_upt_6
- redhat:koji-129949568-aarch64-kernel-64k_upt_6
- redhat:1707109024-ppc64le-kernel_upt_18
- redhat:koji-130442758-aarch64-kernel-64k_upt_6
- redhat:koji-126942713-aarch64-kernel-64k_upt_6
- redhat:1592388569-aarch64-kernel-64k_upt_19
- redhat:koji-126965714-ppc64le-kernel_upt_12
- redhat:koji-126965714-aarch64-kernel-64k_upt_6
- redhat:1593667542-ppc64le-kernel_upt_8
- redhat:1610403851-ppc64le-kernel_upt_7
- redhat:koji-127641511-aarch64-kernel-64k_upt_6
- redhat:1618796740-aarch64-kernel-64k_upt_25
- redhat:1620200259-aarch64-kernel-64k_upt_19
- redhat:koji-127862062-ppc64le-kernel_upt_12
- redhat:koji-127862062-aarch64-kernel-64k_upt_6
- redhat:1732228635-aarch64-kernel-64k_upt_28
- redhat:1732307255-aarch64-kernel-64k_upt_68
- redhat:1736522618-aarch64-kernel-64k_upt_18
- redhat:1736507484-aarch64-kernel_upt_20
- redhat:1738799082-aarch64-kernel-rt-64k_upt_18
- redhat:1738799119-ppc64le-kernel_upt_10
- redhat:koji-130814469-aarch64-kernel-64k_upt_6
- redhat:koji-130814469-aarch64-kernel_upt_6
- redhat:1741035423-ppc64le-kernel_upt_16
- redhat:koji-130946606-aarch64-kernel-64k_upt_6
- redhat:1745508299-ppc64le-kernel_upt_17
- redhat:1745508299-ppc64le-kernel_upt_17
- redhat:koji-133463419-x86_64-kernel_upt_6
- redhat:koji-133463419-x86_64-kernel_upt_6
- redhat:koji-130987467-aarch64-kernel-64k_upt_6
- redhat:koji-131343286-aarch64-kernel-64k_upt_6
- redhat:1766309500-aarch64-kernel-64k_upt_71
- redhat:koji-131564191-aarch64-kernel-64k_upt_6
- redhat:koji-131564191-aarch64-kernel-64k_upt_6
- redhat:koji-131646726-aarch64-kernel-64k_upt_6
- redhat:koji-131679199-x86_64-kernel_upt_6
- redhat:koji-131861103-x86_64-kernel_upt_6
- redhat:koji-131861103-x86_64-kernel_upt_6
- redhat:koji-131910167-x86_64-kernel_upt_6
- redhat:koji-132433399-x86_64-kernel_upt_6
- redhat:koji-132682609-x86_64-kernel_upt_6
- redhat:koji-132682609-x86_64-kernel_upt_6
- redhat:koji-132733158-x86_64-kernel_upt_6
- redhat:koji-133013980-x86_64-kernel_upt_6
- redhat:koji-133624968-x86_64-kernel_upt_6
- redhat:koji-133368318-x86_64-kernel_upt_6
- redhat:koji-133873019-x86_64-kernel_upt_6
- redhat:koji-134361516-x86_64-kernel_upt_6
- redhat:koji-134772687-x86_64-kernel_upt_6
- redhat:koji-135166239-x86_64-kernel_upt_6
- redhat:koji-135437850-x86_64-kernel_upt_6
- redhat:koji-135704936-x86_64-kernel_upt_6
- redhat:1587400690-ppc64le-kernel_upt_12
- redhat:1587473818-ppc64le-kernel_upt_8
```

```
$ kci-dev results issue --id maestro:d8317f5d8e192f5277f02847d9c2dbc0a47c1888
Issue information:
- Issue ID: maestro:d8317f5d8e192f5277f02847d9c2dbc0a47c1888
  origin: maestro
  version: 1
  comment:  suggest parentheses around comparison in operand of ‘|’ [-Werror=parentheses] in drivers/misc/mediatek/slbc/slbc_mt8196.o (drivers/misc/mediatek/slbc/slbc_mt8196.c) [logspec:kbuild,kbuild.compiler.error]
  field_timestamp: 2025-08-06T10:58:37.609586Z
  culprit_code: True
  culprit_tool: False
  culprit_harness: False

Associated builds:
- maestro:688395cef2919f099dfc1fd2
- maestro:6886e5dff2919f099d049c05
- maestro:6887fd83759fd8fd442bccf9
- maestro:6888e5a1759fd8fd442ff66b
- maestro:688abe85759fd8fd4437b1d1
- maestro:688b21af759fd8fd4438a7a3
- maestro:688c55fe236da8c8a0d0dbd3
- maestro:688c8d68236da8c8a0d1f505
- maestro:68906303236da8c8a0d77cfa
- maestro:6892e6a079d5d217e545652c

Associated tests:
No associated tests found

```

aliceinwire commented on kcidev/subcommands/results/parser.py:
```diff
@@ -939,17 +939,56 @@ def format_item_json(regression):
-def cmd_issue_list(data, use_json):
-    """Display KCIDB issues"""
+def print_issues(issues, use_json):
+    """Method to print a list of issues with formatting"""
```
This is a function not a method, you need a class for make this a method

JenySadadia commented:
> Hi Jeny, thanks for these changes. I'll add some comments:
> 
Thank you for the feedback.

> ## `kci-dev results issues`
> * highlight comment as the title - first line of the data. In that case I think we can just omit "comment:"
> * add a link to the dashboard.
> * add tree/branch of the first incident (if there is no incident, we can ignore that issue for now)
> * new issues should have a NEW tag in red.

Addressed all the comments except the last one.
Here we are listing down all the issues. The issue will have tree/branch of the first incident. Hence, this tag isn't applicable here. The tag can be applied if we separate issues based on tree/branch.

> ## `kci-dev results issues --missing`
> * add link to the dashboard, instead of issue id directly
> * then, in front of the link add "(status: )", so we know the status before clicking on the link
> 

Done.

> ## `kci-dev results issues --new <>`
> * should use the same output form as kci-dev results issues`. If we add tree/branch of the first incident there, then we are ready to just pass the a list of issues to the print function.

Done.

aliceinwire commented:
> > > 1. `kci-dev results issues  --giturl https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git --branch master`
> > > 
> > > is not filtering by tree/branch
> > 
> > 
> > I think this was already discussed on the etherpad.
> > ```
> > 3. Add --giturl, --branch --commit (or --latest) parameters to `results issues`, so we can filter then per tree/branch/commit
> > 
> > > (Jeny) Issues are independent of git branch, URL, or commit. Hence, can't be filtered by these options. If we filter issues based on incidents, issues without any incidents associated to it, will not be listed.
> > 
> > > (Gus) Indeed. However, I mean filtering by the info on the incident. Maybe we can park this as just create a GH issues for the time being, then wait the work on the Dashboard DB to solidify. We will do a issues refactor when that happens.
> > 
> > > (Jeny) Ok, I'll create a GH issue. Thanks.
> > ```
> 
> As per our latest conversation, I'll handle issue filtering from kci-dev side for now.
> 
> > > 2. if I add --new to the line above without mentioning the --commit, it should give a error telling the user that the commit is missing
> > 
> > 
> > Done.



> > > > 1. `kci-dev results issues  --giturl https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git --branch master`
> > > > 
> > > > is not filtering by tree/branch
> > > 
> > > 
> > > I think this was already discussed on the etherpad.
> > 
> > 
> > what etherpad?
> 
> This PR is based on the feedback I received from @padovan Here is the link to it https://etherpad.collabora.com/p/kci-dev_results_issues



> > > 1. `kci-dev results issues  --giturl https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git --branch master`
> > > 
> > > is not filtering by tree/branch
> > 
> > 
> > I think this was already discussed on the etherpad.
> > ```
> > 3. Add --giturl, --branch --commit (or --latest) parameters to `results issues`, so we can filter then per tree/branch/commit
> > 
> > > (Jeny) Issues are independent of git branch, URL, or commit. Hence, can't be filtered by these options. If we filter issues based on incidents, issues without any incidents associated to it, will not be listed.
> > 
> > > (Gus) Indeed. However, I mean filtering by the info on the incident. Maybe we can park this as just create a GH issues for the time being, then wait the work on the Dashboard DB to solidify. We will do a issues refactor when that happens.
> > 
> > > (Jeny) Ok, I'll create a GH issue. Thanks.
> > ```
> 
> As per our latest conversation, I'll handle issue filtering from kci-dev side for now.
> 

Filtering issues will be handled on this PR or a different PR?

aliceinwire commented:
> I'll make changes to this PR.

ok, thanks

JenySadadia commented:
Hi @padovan 

Here is the sample output with issue filtering working:
```
$ kci-dev results issues --giturl https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git --branch master --commit next-20250820 --origin maestro
resolved to: 5303936d609e09665deda94eaedf26a0e5c3a087
next/master:5303936d609e09665deda94eaedf26a0e5c3a087
-  the frame size of 1448 bytes is larger than 1024 bytes [-Werror=frame-larger-than=] in drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn10/dcn10_hwseq.o (drivers/gpu/drm/amd/amdgpu/../display/dc/hwss/dcn10/dcn10_hwseq.c) [logspec:kbuild,kbuild.compiler.error]
  https://dashboard.kernelci.org/issue/maestro:4d4120bf99fecc5f852604382b7c406b99440116
  origin: maestro
  version: 1
  field_timestamp: 2025-08-20T05:03:27.970827Z
  culprit_code: True
  First incident seen on next/master

-  ‘pinmux_generic_get_function_count’ undeclared here (not in a function); did you mean ‘pinmux_generic_free_functions’? in drivers/pinctrl/qcom/pinctrl-msm.o (drivers/pinctrl/qcom/pinctrl-msm.c) [logspec:kbuild,kbuild.compiler.error]  NEW
  https://dashboard.kernelci.org/issue/maestro:4466df3c987ba2945eca05fcb2f3c22e97d8636a
  origin: maestro
  version: 1
  field_timestamp: 2025-08-20T04:52:00.879437Z
  culprit_code: True
  First incident seen on next/master

-  Bootloader did not finish or kernel did not start. [logspec:generic_linux_boot,maestro.linux.kernel.boot]
  https://dashboard.kernelci.org/issue/maestro:da694c56147298d223ee432ad8d6a8ee311b773a
  origin: maestro
  version: 1
  field_timestamp: 2025-08-20T09:41:12.706877Z
  culprit_code: True
  First incident seen on net-next/main

-  sleeping function called from invalid context at kernel/locking/rwsem.c:1589 [logspec:generic_linux_boot,linux.kernel.bug]  NEW
  https://dashboard.kernelci.org/issue/maestro:4a7604ad2b208f9142109c871e17a6453ee9ff9c
  origin: maestro
  version: 1
  field_timestamp: 2025-08-20T07:25:46.101792Z
  culprit_code: True
  First incident seen on next/master

-  shift-out-of-bounds: shift exponent -1 is negative [logspec:generic_linux_boot,linux.kernel.ubsan]
  https://dashboard.kernelci.org/issue/maestro:b91eba41d9d0281c086ee574a82bdee035760751
  origin: maestro
  version: 1
  field_timestamp: 2025-08-20T07:25:46.101792Z
  culprit_code: True
  First incident seen on None/None
```
Note: New issues will have a `NEW` tag alongside the issue title.

aliceinwire commented:
> > @JenySadadia sounds good. The 2 issues that don't have next/master in the first incident still have incidents on the tree.
> > As for the NEW tag, let's add it to the beginning of the line:
> > `[NEW] sleeping function called from invalid context at kernel/locking/rwsem.c:1589 [logspec:generic_linux_boot,linux.kernel.bug]`
> > With `[NEW]` in green or red
> 
> Done. Updated the tag to have the red color.

As we are showing issues, I also think that red is better
